From b3053fa0f83a55bb9fb8148eab51089171da21cb Mon Sep 17 00:00:00 2001
From: HATAYAMA Daisuke <d.hatayama@fujitsu.com>
Date: Tue, 14 Feb 2023 22:10:13 -0500
Subject: [PATCH] sulogin: use DEFINE_MAIN_FUNCTION()

Let's use DEFINE_MAIN_FUNCTION() as the other commands for consistency.

This commit should be no functional change.

(cherry picked from commit 2ffbf44344983d64949e032e74edb19c48b16cc0)

Related: #2169959
---
 src/sulogin-shell/sulogin-shell.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/sulogin-shell/sulogin-shell.c b/src/sulogin-shell/sulogin-shell.c
index 5648dfd83b..e3601de8d4 100644
--- a/src/sulogin-shell/sulogin-shell.c
+++ b/src/sulogin-shell/sulogin-shell.c
@@ -14,6 +14,7 @@
 #include "def.h"
 #include "env-util.h"
 #include "log.h"
+#include "main-func.h"
 #include "process-util.h"
 #include "signal-util.h"
 #include "special.h"
@@ -86,7 +87,7 @@ static void print_mode(const char* mode) {
         fflush(stdout);
 }
 
-int main(int argc, char *argv[]) {
+static int run(int argc, char *argv[]) {
         const char* sulogin_cmdline[] = {
                 SULOGIN,
                 NULL,             /* --force */
@@ -115,5 +116,7 @@ int main(int argc, char *argv[]) {
                 r = start_default_target(bus);
         }
 
-        return r >= 0 ? EXIT_SUCCESS : EXIT_FAILURE;
+        return r;
 }
+
+DEFINE_MAIN_FUNCTION(run);
